src/main/kotlin/com/autonomousapps/subplugin/ProjectPlugin.kt

```diff
@@ -487,9 +487,13 @@ internal class ProjectPlugin(private val project: Project) {
       setCompileClasspath(
         configurations[dependencyAnalyzer.compileConfigurationName].artifactsFor(dependencyAnalyzer.attributeValueJar)
       )
+      setRuntimeClasspath(
+        configurations[dependencyAnalyzer.runtimeConfigurationName].artifactsFor(dependencyAnalyzer.attributeValueJar)
+      )
       buildPath.set(buildPath(dependencyAnalyzer.compileConfigurationName))
 
-      output.set(outputPaths.artifactsPath)
+      output.set(outputPaths.compileArtifactsPath)
+      outputRuntime.set(outputPaths.runtimeArtifactsPath)
```

Comment:
Just as suggestion, if `setCompileClasspath` was renamed `setClasspath` you could use the same task to register `ArtifactsReportTask` twice with different classpaths and output paths so each configuration could be run independently and/or in parallel:

```kt
val artifactsCompileReportTask = tasks.register<ArtifactsReportTask>("artifactsReportCompile$taskNameSuffix") {
      setClasspath(
        configurations[dependencyAnalyzer.compileConfigurationName].artifactsFor(dependencyAnalyzer.attributeValueJar)
      )
      buildPath.set(buildPath(dependencyAnalyzer.compileConfigurationName))
      output.set(outputPaths.compileArtifactsPath)
}

val artifactsRuntimeReportTask = tasks.register<ArtifactsReportTask>("artifactsReportRuntime$taskNameSuffix") {
      setClasspath(
        configurations[dependencyAnalyzer.runtimeConfigurationName].artifactsFor(dependencyAnalyzer.attributeValueJar)
      )
      buildPath.set(buildPath(dependencyAnalyzer.runtimeConfigurationName))
      output.set(outputPaths.runtimeArtifactsPath)
}

// parent task that runs both
val artifactsReportTask = tasks.register("artifactsReport$taskNameSuffix") {
    dependsOn(artifactsCompileReportTask, artifactsRuntimeReportTask)
}
```